src/open_producten/producttypen/management/commands/load_upl.py

```diff
@@ -1,68 +1,137 @@
-from ..parsers import CsvParser
-
 
 @dataclass
 class UniformProductName:
     name: str
     uri: str
 
 
+def _check_if_csv_extension(path: str):
+    _, extension = os.path.splitext(path)
+    file_format = extension[1:]
+
+    if file_format != "csv":
+        raise CommandError("File format is not csv.")
+
+
 class Command(BaseCommand):
 
     def __init__(self):
-        self.help = "Load upn to the database from a given XML/CSV file."
-        self.parser = CsvParser()
+        self.help = (
+            "Load upn to the database from a given local csv file or csv file url."
+        )
         super().__init__()
 
     def add_arguments(self, parser):
         parser.add_argument(
-            "filename",
-            help="The name of the file to be imported.",
+            "--file",
+            help="The path to the csv file to be imported.",
+        )
+        parser.add_argument(
+            "--url",
+            help="The url to the csv file to be imported.",
         )
 
     def handle(self, **options):
-        filename = options.pop("filename")
+        file = options.pop("file")
+        url = options.pop("url")
+
+        if file and url:
+            raise CommandError("Only one of --file or --url can be specified.")
 
-        self.stdout.write(f"Importing upn from {filename}...")
+        if not file and not url:
+            raise CommandError("Either --file or --url must be specified.")
+
+        self.stdout.write(f"Importing upn from {file if file else url}...")
 
         try:
-            data = self.parser.parse(filename)
-            upn_objects = [
-                UniformProductName(name=entry["UniformeProductnaam"], uri=entry["URI"])
-                for entry in data
-            ]
-            created_count = self.load_upl(upn_objects)
-
-        except KeyError as e:
-            raise CommandError(f"{str(e)} does not exist in csv.")
+            if file:
+                created_count, update_count, removed_count = self._parse_csv_file(file)
+            else:
+                created_count, update_count, removed_count = self._parse_csv_url(url)
+        except CommandError:
+            raise
         except Exception as e:
-            raise CommandError(str(e))
+            raise CommandError(f"Something went wrong: {str(e)}")
+        self.stdout.write(
+            "Done\n"
+            f"Created {created_count} product names.\n"
+            f"Updated {update_count} product names.\n"
+            f"{removed_count} product names did not exist in the csv."
+        )
+
+    def _parse_csv_file(self, file: str):
+        _check_if_csv_extension(file)
+
+        with open(file, encoding="utf-8-sig") as f:
+            data = csv.DictReader(f)
+            return self._load_upl(data)
+
+    def _parse_csv_url(self, url: str):
+        _check_if_csv_extension(url)
 
-        self.stdout.write(f"Done ({created_count} objects).")
+        response = requests.get(url)
+        if response.status_code != 200:
```

Comment:
Not sure if you intended it, but this will not catch 400 or 500 codes, only different 200 and 300 codes

Reply:
requests throws exceptions for those codes